New Timetable

tomasr8 opened the pull request.

tomasr8 committed:
Fix z-index issues when resizing child entries

tomasr8 committed:
Prevent resizing beyond the end of the parent block

ThiefMaster committed:
Fix linter errors

AjobK committed:
Cancel dragging with escape (#6604)

AjobK committed:
Change buttons to divs (#6605)

AjobK committed:
Disable resizing entries further than children (#6699)

* disallow shrinking more than child elements

* disable selection on drag

* unbehead changed files

* unbehead all timetable files

duartegalvao committed:
Use event's time boundaries (#6635)

* Use event's time boundaries

* Apply suggestions and code improvements

* minHour--

---------

Co-authored-by: Tomas R. <tomas.roun8@gmail.com>

tomasr8 committed:
contribution -> contrib

tomasr8 committed:
Set the parentId when dropping entries on the calendar

tomasr8 committed:
Set the parentId when dropping entries on a block

tomasr8 committed:
Add a param to conditionally layout block children

tomasr8 committed:
WIP tests

tomasr8 committed:
Reactify timetable dialogs

Co-authored-by: Duarte Galvão <duarte.galvao@cern.ch>

tomasr8 committed:
Fix a bug in onDrop which sometimes duplicated an entry

tomasr8 committed:
Do not store width/left CSS props, but compute them on demand

We alrady store column & maxColumn from the CSS width and left
offset can be easily computed when needed.

tomasr8 committed:
Add more layout tests

tomasr8 committed:
Add some code comments

tomasr8 committed:
Fix key error

tomasr8 committed:
Fix a bug in layout

tomasr8 committed:
Add basic support for scheduling contributions

AjobK committed:
unbehead

AjobK committed:
Timetable click drag creation (#6733)

* resize by dragging

* open create contrib modal

* initial modification and usage of custom contrib modal

* basic dialogues

* fixed sizing issue and startdate

* escape to cancel new entry

* add default startdate

* fix date error

* rm redundant code and keep date when switching forms

* extract sessionform logic

* functional data passage between forms

* session block and session in one form

* colored sessions selector

* support multiple forms and pass data to back-end

* functional sessionblock create

* contrib creation

* add from backend data to redux

* rm unwanted recommendation in extensions

* session color picker to tsx

* draft of cleaner entry creation

* better entry response formatting

* prepare edit functionalities

* fix creation timezone issue, add color to break, etc

* disable sessioblock creation when no session

* rm consolelog

* enable submitting again

* Mapping to person links data

* clean up and allow person links for sessionblock and contribution

* rm obsolete code and fix title issue for session blocks

* limit date choice and fix focusbutton nullpointer

* cleanup and going through todos

* fix lint issues

* moved personlinks mapping fn and fixed bug in contrib form

* rm optional chaining in favour of linter

* fix ruff errors

* minor pr comments

* cleanup and fix date issues

* unbehead

* isort schemas file

tomasr8 committed:
WIP: timetable popups

tomasr8 committed:
Deselect entry with Escape and when scrolling

tomasr8 committed:
Improve entry popups

tomasr8 committed:
Expose old timetable

tomasr8 committed:
Only show sessions with contribs

tomasr8 committed:
Comment out popup toggle

tomasr8 committed:
Wire up deleting breaks

tomasr8 committed:
Rename DELETE_ENTRY

tomasr8 committed:
Wire up deleting blocks

tomasr8 committed:
Do not propagate popup click events

tomasr8 committed:
Hide popup when dragging

tomasr8 committed:
Optimize checking an entry is selected

tomasr8 committed:
Remove some unused code

tomasr8 committed:
Add DESELECT_ENTRY

tomasr8 committed:
Fix block title display

tomasr8 committed:
Prevent selecting entry on dragEnd

tomasr8 committed:
Fix copy-paste error

tomasr8 committed:
Deselect entries when dragging starts

tomasr8 committed:
Fix incorrect ref usage

tomasr8 committed:
Wire up unscheduling contribs

AjobK committed:
Allow editing timetable entries (#6942)

* replaced state management for edit

* clean up popup (dry), bugfixes, made create modal fit editing

* refactoring, remove wet

* fix person links

* fix users addition to sessionblocks and contribs

* post-rebase changes

* merge entry components

* fix double tooltip

* Disable right click where unnecessary

* cleanup entry.tsx

* pass previously ignored fields for contrib on creation

* additional fields sessionblock

* break additional

* Board number and program code visible again in edit

* new serializer

* pass avatarUrl

* cleanup of modal

* fetch entry data on edit and fix creation

* fix location data, draft

* rm console logs

* process_PATCH marshmallow progress

* passing patch without saving to db

* functional contrib patch

* functional load data on click for contrib

* functional edit of contrib

* functional edit with state update

* break get and patch

* functional edit dialogues opening

* creation from timetable functional again

* cleanup and fixed description entry in breakform

* edit working without person links, creation still has person links

* isort

* edit right after creation without crash

* fix colors and location fields

* remove prints and logs

* fix ruff issues

* more ruff

* adrian comments

* using rh_args decorator instead of experimental

* remove unnecessary, cleanup and make independent forms functional

* isort schemas.py

* rm unnecessary classes

* clean-up

* rm unintended newline

* rm unintentional additional field to session schema

* child ids working noiw

* fix same-id bug

* rm unused

* rm unused partial flags

* tomas comments pr

tomasr8 committed:
Fix a really strange webpack error

tomasr8 committed:
Apply ruff fixes

tomasr8 committed:
Remove references to marshmallow_enum

tomasr8 committed:
Fix deprecation warning

AjobK committed:
Use real data in `EntryPopup` (#6958)

* real data in entrypopup

* make code more readable and remove redundant things

* cleanup serializer

AjobK committed:
Revert "Use real data in `EntryPopup` (#6958)"

This reverts commit bd8519a37c28e9b9c45bf906acccc3026a2d2ea4.

AjobK committed:
Fix double dialogue (#6965)

* make serializer more readable, add uniqueid, remove redundant code

* separated uniqueID and objectID

* rm unused id props

* fix unscheduled ids and move modified legacy event serializer to new

* fix child type being passed on edit

marinaej committed:
Implement entry deletion feature (#6955)

* Implement entry deletion feature

* fix delete function

* Change icon for contributions

* Add tooltip & button color

* Fix actions request

* Fix actions form ajobs comments

* fix imports and variable names

* remove redundant  error handling

* remove try catch

marinaej committed:
Fix delete for unscheduled contributions (#6973)

AjobK committed:
Change draggable id type to string (#6970)

AjobK committed:
Use real data in timetable popup (#6966)

* real data in entrypopup

* make code more readable and remove redundant things

* cleanup serializer

* cleanup of interfaces, fixed crashing dialogue and translated strings

* fixed merge conflict

* semantic issue in mapping fn

* show child info

* move locationparent interface

* fix child data

* interfaces and attachments showing for children

* make personLinks non optional

tomasr8 committed:
Remove unused popup toggle code (#6982)

tomasr8 committed:
Remove extra onclick and make prettier work (#6983)

* Make prettier happy & reformat code

* Remove extra onclick

AjobK committed:
fix double mapping bug

AjobK committed:
Make timetable scrolling while dragging smoother (#6986)

* change refresh rate and speed

* mv const pos

tomasr8 committed:
Improve the timetable auto-scroll (#6985)

* Disable mouseenter for toolbar day selector

* Disable scroll when mouse is outside scrollable area

* Clamp values instead of disabling scroll

renefs committed:
Remove color picker from child breaks (#6992)

* Remove color from nested breaks

* Apply suggestion from @tomasr8

Co-authored-by: Tomas R. <tomas.roun8@gmail.com>

* Extract parentId assignment on mapTTDataToEntry

* Update indico/modules/events/timetable/client/js/TimetableManageModal.tsx

Co-authored-by: Tomas R. <tomas.roun8@gmail.com>

---------

Co-authored-by: Tomas R. <tomas.roun8@gmail.com>

tomasr8 committed:
Show the time only when unscheduled entry is over the calendar

tomasr8 committed:
Fix drag&drop on an empty session block (#6995)

* Fix drop on empty session blocks

* Update indico/modules/events/timetable/client/js/ContributionEntry.tsx

Co-authored-by: Ajob Kustra <ajobkustra.p@gmail.com>

---------

Co-authored-by: Ajob Kustra <ajobkustra.p@gmail.com>

tomasr8 committed:
Prevent unnecessary re-renders when selecting session blocks containing children (#6984)

* Add a utility hook to trace props

* Keep onMouseDown identity when draggables change

* Update indico/modules/events/timetable/client/js/dnd/dnd.tsx

Co-authored-by: Ajob Kustra <ajobkustra.p@gmail.com>

* Add a todo comment

---------

Co-authored-by: Ajob Kustra <ajobkustra.p@gmail.com>

tomasr8 committed:
Fix typo

tomasr8 committed:
Fix enum name

tomasr8 committed:
Return more data in unscheduled contribs (#6999)

tomasr8 committed:
Wire up scheduling top-level entries (#7005)

* Wire up scheduling top-level entries

* Fix use_kwargs location

tomasr8 committed:
Clean up unused code (#7007)

tomasr8 committed:
Wire up scheduling child entries (#7006)

tomasr8 committed:
Fix session block title display (#7009)

AjobK committed:
Remove timetable group console log

tomasr8 committed:
Clean up more unused code (#7011)

tomasr8 committed:
Rename timetable redux store

ThiefMaster committed:
Clean person data in all PersonLinkField instances

In the react versions we still passed the whole data from the current
user in case of "add myself", sending lots of unnecessary crap to the
server

ThiefMaster committed:
Add util to get locations w/ rooms

ThiefMaster committed:
Fix querying room/location data

This was doing a rooms x rooms cross join, resulting in a hugely
inefficient query returning O(n^2) rows.

ThiefMaster committed:
Support context-based SeachToken in PersonLinkField

ThiefMaster committed:
Provide search token to person link fields

ThiefMaster committed:
Add missing linebreak at EOF

AjobK committed:
fix scroll on old timetable

marinaej committed:
Add dragging for new timetable (#6990)

* Add drag feature for new timetable

* minor adjustements

* remove console.log

* review fixes

* Optimize handleDropOnBlock and moveEntry functions

* Optimize layoutAfterDrop

* Optimize moveEntry function

* Remove comments

ThiefMaster committed:
Fix indentation/formatting of BreakForm

micsucmed committed:
Add new state for current day (#7014)

* Add new state for current day

* Fix state being overwritten by resizeWindow

* Add new state to toolbar use

* add localstorage currentdate, rename currentday to currentdate and cleanup

* rm redundant code and cleanup

* rm unnecessary whitelines

* Reorder moment import

* simplify localstorage fn

---------

Co-authored-by: AjobK <ajob.edward.kustra@cern.ch>

tomasr8 committed:
Serialize default contrib duration (#7035)

AjobK committed:
Timetable refactor toolbar (#7008)

* fix

* progress on replacing redux toolbar logic

* hide scrollbar

* increased padding to completely remove scrollbarr

* make 'right' and 'left' go by increments of one, fixed page-content issue

* separated toolbars, expand feature, redesign

* store current dt locally

* fix bug for events of less than 24h

* scrollto day

* jump by page

* rm console.log

* indent, abandon sticky month idea, show month on active

* rm unused offset and other redux states in toolbar

* rm go to end and start, change double arrow to page basis, combien fns

* Cleanup, fix issues with dropdown displays, redesign toolbar

* get rid of all timeslots option

* rm dropdown header

* conditional days rendering and reintroduce draft warning. two button options for draft to evaluate

* add publish draft button and remove old warning

* clean up publication button code

* add comment to reviewchanges button

* use publishstateswitch and remove old approach with button

* change publish button icon and reposition

* add translate to dropdown

* rm todo

* rm ibutton changes

* rm margintop on old timetable to prevent draftmode warning being covered

* initial fix for adding entry through toolbar

* fix last day bug, rm display mode code

* rm dead openModal code and console.log

* fix date issue in addNewEntry on toolbar

* rm publicationmodal divider

* use default contrib minutes selector

* rename regular users to event participants, tsxify toggle (it was only non tsx file in folder)

* get rid of useeffect warning, apply exit fullscreen naming to expand toggle

* rm icondisplay mode in publicationstateswitch

* start at 8 am for new entries

* good english

* change typing for ref

tomasr8 committed:
Fix handleDragEnd

tomasr8 committed:
Optimistically update the timetable UI and synchronize network requests (#7028)

* Disable eslint class member rule for ts files

* Optimistically update timetable UI

* Update indico/modules/events/timetable/client/js/actions.ts

Co-authored-by: Ajob Kustra <ajobkustra.p@gmail.com>

* Update indico/modules/events/timetable/client/js/actions.ts

* Update indico/modules/events/timetable/client/js/actions.ts

---------

Co-authored-by: Ajob Kustra <ajobkustra.p@gmail.com>

ThiefMaster committed:
Reformat timetable codebase with biome

ThiefMaster committed:
Remove obsolete files

ThiefMaster committed:
Make @typescript-eslint/no-explicit-any a warning

tomasr8 committed:
Remove more unused code (#7044)

* Remove unused components

* Remove unused code

tomasr8 committed:
Clean up linter errors (#7045)

* Clean up linter errors

* Fix dependency cycle

tomasr8 committed:
Check if daysBarRef is undefined

micsucmed committed:
Timetable create session block child (#7037)

* Get the session for the current session block

Add some relevant information for child creation

Trigger creation modal when selecting to create a child

* Filter session block tab when creating child

* Remove completed TODO's

* Add session block id to payload

* Implement default location when creating an entry

* Ensure location's data immutability

* Update TODO's

* Create new break child

* Improve selector readability

* Remove print

* Replace parentId for sessionBlockId

* Simplify Break schema for session_id serializer

* Simplify Break schema serilizer of session_block_id

Fix unschedule and delete functions for new sessionBlockId

* Make session_block_id consistent on create Break requests

* Rename default contribution time for sessions

* Refactor entry type comparisons

* Simplify child addition on reducer

* Add axios erros handler

* Add no_autoflush to contrib creation

tomasr8 committed:
Fix quotes

tomasr8 committed:
Update eslint import resolver

micsucmed committed:
Timetable reschedule edit entries (#6977)

* Assign new date on schedule edit

* Test new date logs

* Fix start day edit when changing day

Fix children disappearing

* Revert children schema

* Replicate date changes for children

* Fix mutability arrays

Clarify variable names

* Add children day shift as a util function

* Refactor shift entries util

* Assign new variable for the draftEntry

* Remove unused import

* Improve immutability

* Use currentDate state to simplify

* Remove unclear TODO

* Fix date overwrite

* Fix top level contribution editing

* Remove whitespaces

* Remove unused imports

micsucmed committed:
Display session title correctly (#7046)

* Check for timedelta and shift children only on editing

* Fix undefined session title and colors

* Add missing trailing semicolon

* Simplify getting session title

Correct entry type comparisons

* Remove undefined import

micsucmed committed:
Remove extend_parent from create_contribution

AjobK committed:
fix scroll on old timetable again

AjobK committed:
semicolon on scroll fix timetable

AjobK committed:
Refactor usage of colors in timetable (#7060)

* provide better color properties through serializer and preprocessor

* make entry colors consistent with semantics of serializer

* replace textColor and backgroundColor with colors for unscheduled

* fix color for coloreddot in entrypopup

* replace mapping for colors when communicating with db

* move colors assignment in mapper

* glass effect draftentry

* rm print

* fix styling on draftentry

* rm now unnecessary isChild

* resolve lint errors

* fix biome error

* fix more biome errors

* adrian comments

* biome comments

* added back forgotten commas

* fixed last semicolon error in biome

renefs committed:
Remove required title from SessionBlockFormFields

Handle naive dates in EventTimezoneDateTimeField

micsucmed committed:
Add session title to session block schema

Simplifies mantaining session title when updating/creating session blocks

micsucmed committed:
Update child information when edited

AjobK committed:
Disable interaction outside of event times and implement initial scroll position (#6962)

* Add session title to session block schema (#7053)

Simplifies mantaining session title when updating/creating session blocks

* limits gradient

* cleaner gradient

* functional limits

* fix out-of-interaction-field bug

* todo comments

* fix end duration issue

* fix enddt bug in mousemove

* cleanup and module on 24hr

* PR comments

* initial scroll position

* fix scrolltop bug for single day events

* disabling areas from dropping and scrolling, wip

* functional scroll restrict, sometimes breaks on scrollend

* functional scroll, but somewhat unpredictable bottom limit

* format

* fix bug where gradient was day-scoped and not full date

* scrollTo only when daysbarref current exists

* rm scroll limits because the implementation sucks

* add defualt contribution times and move pixellimits etc to state

* get rid of non-24 hour view and scroll to first element or event start dt

* rm unused imports

* use getDateKey

* fix broken usage of defaultcontrib

* add day and hour size vars for easy calculations

* unused import

* add min scroll hour

* fix empty moment.min bug

* cleanup

* replace createRestrictToCalendar fn with limits based one

* rm unused const

* move draftEntry selector

* move some other selectors and state vars

* fix drag and create bug

* cleanup code and make more readable

* biome prettier

* fix startdt bug of stepsize 20 min which should be 5 min

* adrian comments

* formatted modifiers.ts

* cleanup

* cleanup var names

* tomas PR comments

* rm unused imports and strange biome formatting

* lint issues

* tomas comment about refs

* fix reversed logic limits and clean up timetable.tsx

* fix biome lint error

* fix rebase conflict

* limitsDelta dependency warning

* clarification comment

* useeffect workaround shallow comparison

---------

Co-authored-by: Michel Succar Medina <53895656+micsucmed@users.noreply.github.com>

AjobK committed:
remove overwriting of colors by parent colors, use transitive value instead

AjobK committed:
fix linting issues and opening edit modal nullpointer

ThiefMaster committed:
Send partial updates + do not lose person links (#7080)

* Fix typo in comment
* Send partial updates when editing entries
* Do not remove person links in the client
* Do not delete person links if data is missing

AjobK committed:
Fix toolbar and limits miscalculations (#7076)

* make sure dates are copied and toolbar takes date differences into account properly when moving to other dates

* cleanup util fn and write tests

* add semicolon

* rm unused attr

* biome error fix

ThiefMaster committed:
Fix location inheritance issues (#7081)

* Fix location parent when editing entries

* Pass schema validation errors to final-form

* Clear location fields when disabling inheritance

* Add DebugFormSpy component

* Fix FinalLocationField value equality check

* Remove weird locationData conversion roundtrip

* Fix location field when creating new entries

* Remove legacy location data

* Move location schemas to a less awful location

Still not super happy with the place, but better than importing it from
the sessions module...

tomasr8 committed:
Remove unused selector

tomasr8 committed:
Reduce ts errors in reducers.ts

tomasr8 committed:
Reduce ts errors in reducers.ts

tomasr8 committed:
Remove unused code (#7085)

tomasr8 committed:
Fix TS errors in reducers.ts (#7086)

tomasr8 committed:
Fix TS errors in dnd

tomasr8 committed:
Fix TS errors in layout.ts

AjobK committed:
fix initial colors for unscheduled without session

tomasr8 committed:
Fix TS errors in layout.spec.ts

tomasr8 committed:
Fix TS errors in UnscheduledContributionEntry.tsx

tomasr8 committed:
Fix TS errors in TimetableManageModal.tsx

tomasr8 committed:
Fix TS errors in Entry.tsx

tomasr8 committed:
Teach TS about the Indico global var

tomasr8 committed:
Fix most TS errors in ContributionForm.tsx

tomasr8 committed:
Fix TS errors and remove unused code

tomasr8 committed:
Update FinalDropdown type defs

tomasr8 committed:
Comment out WeekTimetable to silence a bunch of TS errors

tomasr8 committed:
Remove unused function

tomasr8 committed:
Fix TS errors in BreakForm.tsx

tomasr8 committed:
Fix TS errors in SessionColorPicker.tsx

tomasr8 committed:
Fix TS errors in ContributionForm.tsx

tomasr8 committed:
Fix TS errors in SessionBlockForm.tsx

tomasr8 committed:
Fix TS errors with selected entry selector

tomasr8 committed:
Fix TS errors in preprocess.ts

tomasr8 committed:
Add proper typing for hex colors

tomasr8 committed:
Update final field type defs

tomasr8 committed:
ts-ignore hard-to-fix error

tomasr8 committed:
Add proper typing for session objects

tomasr8 committed:
Temporarily ignore errors in EntryPopup.tsx

tomasr8 committed:
Use default colors for top-level contribs

tomasr8 committed:
Fix TS errors in DayTimetable.tsx

tomasr8 committed:
Fix header

tomasr8 committed:
Remove WeekTimetable.tsx

marinaej committed:
Add resizing for new timetable (#6997)

* Fix resizing for children

* add resizing for entries

* Add post-review fixes

* Remove console logs

* Fix if loop in reducer

AjobK committed:
fix broken entry creation

AjobK committed:
make isChild function return actual boolean value

AjobK committed:
Refactor and redesign timetable entry popup (#7056)

* initial design for entrypopup. still has unique key error and does not properly display session name. no block title info displayed either

* better label and design for popup header

* fix unique key issue

* add colors and cleanup popup

* cleanup colors and actions

* cleanup colors logic and move buttons in entrypopup

* biome error fixes

* rm dead code and fix eslint errors

* prettier and biome fixes

* undo out-of-scope timetable changes

* do not add props if they do not exist in mapper

* cleanup

* ignore no use before define

* cleanup interfaces and rm unused

* fix unused import

* adapt code to get colors inside entrypopup

* minor redesign

* cleanup, remove redundant code

* fix minor rebase bug

* stop propagation on click

* fix biome error

* lint and tsc fixes

* Adrian comments

AjobK committed:
Add session creation and rework sidepanel in timetable (#7097)

* fix scroll on old timetable again

* minor change to sidepanel to make it clearer that the create session button could be there

* i18n

* add session editing and creation (no redux yet) and change individual labels to dropdown selector in sidepanel

* linters

* fix sorting logic

* cleanup, min sidebar width, categorized dropdown entries, proper sorting, functional search and fixed focus bug after selection

* ellipsis and equal styling on selection search

* cleanup order of code

* delete sessions, add redux logic, separate sessionform to be reusable inside timetable

* cleanup reducers to be more efficient, logic for navigating after CRUD operations, remove nasty callbacks

* rm unused blurdropdown, clean up based on linter comments

* rm util fn and its tests after I found an existing alternative. rm redundant logic, fix patch load as it was still sending the full load before

* rename formData to be more consistent

* fix tsc issues, add any to dispatch to handle thunkaction logic for now

* add header

* fix tsc error regarding unscheduledcontrib interface usage

* biome error

* hide headers for empty session sets, fix chrome browser bug

* keep sidepanel closed by default

* rm unused

* better alignment of text in session

marinaej committed:
Timetable drag entries (#7074)

* fix scroll on old timetable again

* Fix color setting when dragging

* Add drag & drop for all types

* Fix color bug & optimize reducers

* remove unnecessary props

* Fix Biome errors

* Fix Biome errors

* Add review changes

* Add review comment fixes

* Fix linting errors

* Fix TS & Biome errors

* Modify layout logic on dragging

* Fix Biome errors

* Refactor function names

* Optimize frontend

* Fix action function return & biome error

* Fix color bug with ChildEntry

* Add review fixes - Adrian

* Fix regression on resize

* Fix dragging from block to top-level

* Add review fixes - Adrian

* Add unit tests for contrib/break move operations

* Disallow timestamps containing (micro)seconds

* Fix break dragging

---------

Co-authored-by: AjobK <ajob.edward.kustra@cern.ch>
Co-authored-by: Adrian Moennich <adrian.moennich@cern.ch>

ThiefMaster committed:
Fix many strict-mode TS typing errors (#7120)

* Add lodash types

* Remove code that doesn't do anything

* Fix typing issues

* Silence some errors outside strict mode

IDs are a mess of number vs string (with the type prefix) mixed
together, and probably half the typing there is incorrect. CBA to fix
this now.

* Fix typing+behavior of draggable `fixed`

It was set on the draggable data, but typed and read on the draggable
itself...

ThiefMaster committed:
Fix cross-module relative imports

ThiefMaster committed:
Fix formatting issues in jsconfig/tsconfig

tomasr8 committed:
Show color picker when editing top-level breaks (#7189)

tomasr8 committed:
Fix creating child breaks (#7190)

tomasr8 committed:
Fix editing child breaks (#7191)

When editing e.g. a child break title, the changes
would only be visible when refreshing the page.

This is because the edit modal makes a new GET request,
which was previously not returning the session_block_id
of the break leading to redux updateEntry action being "lost".

AjobK committed:
fix session_block_id assignment in break create rh

AjobK committed:
Fix contribution child creation and dynamic limits in creation form (#7198)

* add sessionblockid to child contrib

* fix dynamic limits on startdt and duration

* instead of adjusting duration at limits, adjust startdt

AjobK committed:
Optimize drag-create in timetable and add bi-directional drag creation (#7201)

* very basic bi-directional drag creation

* refactored dragging draft entry to be much faster without rerendering, added bidirectional dragging logic, removed redundant remnants from old draft implementation

* hide draft entry when not dragging

tomasr8 committed:
Update timetable when session is deleted (#7211)

tomasr8 committed:
Prevent scheduling outside event start/end time (#7216)

tomasr8 committed:
Preselect session when only exists in the `TimetableManageModal` (#7218)

* Update final-form TS definitions

* Preselect session when only one exists

tomasr8 committed:
Add vertical scroll to Indico sidebar (#7219)

bpedersen2 commented:
Question: Does this already include the equivalent of #6876 ? That is, board-numbers if set are also shown in the detailed view for  poster sessions?

tomasr8 committed:
Fix deleting session blocks (#7223)

tomasr8 committed:
Fix dropping a contrib larger than the destination block (#7226)

AjobK committed:
reset dragging start pos to null

AjobK committed:
First redesign of timetable entries (#7229)

* left border, paddings, select and drag effect, refactored title, icons changed to outline, stylistic changes to entry itself

* refactor and redesign timeline code and children-wrapper

* Tomas comments

* clarifying comment for webkit-box usage

tomasr8 committed:
Consistently sort unscheduled contribs by (sessionId, title) (#7224)

tomasr8 committed:
Fix unscheduled contrib bg color (#7221)

Makes the unscheduled and scheduled contrib bg color the same.
Previously the unscheduled contrib bg color was the session bg color.

Commit:
Happy new year 2026 :fireworks:

marinaej committed:
Fix child entry color not updated when moved to top level (#7253)

* Remove color condition

* Remove sessionId when moved to top level

tomasr8 committed:
Delete/unschedule entries with the <delete> & <backspace> key (#7220)

* Delete/unschedule entries with the <delete> key

* subEntry -> child

* Use EntryType.SessionBlock

* Apple pls

Co-authored-by: Marina EJ <marina.el.jaouhari@cern.ch>

---------

Co-authored-by: Marina EJ <marina.el.jaouhari@cern.ch>

tomasr8 committed:
Fix the way datetime picker start time limits are computed (#7225)

* Add dynamic time limits to session block form

* Fix dynamic limit computation and slightly refactor

Previously we weren't taking into account the current day in
our limit calculation. For example, the minStartDt was being
applied to *all* days even though it should only apply if
the currently selected date is the *same* as minStartDt.
Same for maxEndDt.

* Simplify and add tests

tomasr8 committed:
Fix TS errors in `EntryPopup.tsx` (#7230)

* Fix TS errors in EntryPopup.tsx

* Inline a function used only once

AjobK committed:
Add session block view to timetable (#7289)

* basic state and onclick for selecting session block view

* session block toolbar

* adjust timetable limits to session block

* fix limits bug

* creation of new entries now happens inside sessionblock view and also updates. Toolbar button and click-drag work

* fix resize issue

* drag unscheduled and scheduled entries with sessionblock scope in mind. modified layouting functions

* proper limits in session block view and support dropping unscheduled entries in any block

* default contrib duration inside session block view

* creation of blocks with defaultcontribduration in mind, especially in small blocks

AjobK committed:
Add possibility to move entries with buttons in timetable (#7261)

* snap to top and bottom working, rm useless draggableblock class

* cleanup of solution, separate file and fully functional move and swap, including moving children in state

* better styling and space for move buttons. also removed obnoxious hover title because buttons are self-explanatory

* unbehead

* stylistic changes, more room (temporarily?) for hovering on parent

* calculate entry above and below automatically. allows for disabling/enabling buttons

* disable sorting inside of overlapped entries, implemented o log n algorithm which runs once on state, not for every entry render

* disable swapping with groups or within groups

* rm outdated todo

* tests for new util functions and factory function for entries

* cleanup based on tomas review

* comments and lint

AjobK committed:
Fix timetable move buttons styling (#7321)

* snap to top and bottom working, rm useless draggableblock class

* cleanup of solution, separate file and fully functional move and swap, including moving children in state

* increase size of buttons

* increase button size

ArmindoFlores committed:
Make EntryPopup aware of its boundaries to prevent overflow (#7309)

AjobK committed:
Add ability to drag and create entries when clicking on timeline of timetable (#7297)

* click drag implementation on time gutter

* mv limits gradient

* put limits gradient color in separate var to avoid redundancy

* fix lint issues

marinaej committed:
Timetable / Enable drag and drop of unscheduled contribs anywhere on calendar (#7344)

* Enable drag and dropping unscheduled on calendar

* Add possibility to add unscheduled to a block from another session

AjobK committed:
rm annoying resize handle spacing

AjobK committed:
Add ability to create sessions directly from timetable entry form (#7305)

* rework session select and change search field in finaldropdown

* add sessions through session selector and use random colors for sessions with no colors set

* session select with adapter, creates session on submit of managemodalform

* replace session dot in unscheduled contrib tab

* unbehead

* cleanup, better styling and added label for drafts

* Tomas comments, also error handling change for finding sessions

AjobK committed:
Add poster session block view (#7355)

* basic poster session logic to differentiate entrypopup and on timetable

* base poster session with with contributions and refactored draft entry and poster session block modals to use same hook context

* base modal with read, update and delete for contributions in posters

* rm flag dot

* add creation of contribs in poster block

* cleanup and rm room booking accidentally pushed code

* add comment explaining modalprovider

* fix tsc errors

* fix test

* unbehead

* fix empty session bug

* tomas comments and tsc errors

* if block.title string is empty, different translation in modal

* change if statement to conditional chaining for onClose

AjobK committed:
Remove Timetable copied values in favor of transitive values (#7363)

* rm non-transitive values (parent, sessionTitle), fix children update bug, lowercase color picker colors (causes errors otherwise in getColor fn)

* fix session block view

* rename props interface and add two props to the it

* fix creation of poster contributions regression

AjobK committed:
fix person links lastName error

ThiefMaster committed:
Make event_id dump-only

Otherwise creating contributions from the timetable fails because we
never send an event_id

ThiefMaster committed:
Fix linter errors (#7436)

AjobK committed:
Rewrite timetable mappers and manage modal (#7412)

* rewrite mapper to allow partial mapping and make models more consistent. Remove multiple handle functions from timetablemanagemodal, remove unnecessary event_id passage from backend schema, replace color mapping fn with partial mapper, add tests to ensure mappers work properly

* fix tsc and lint issues

* rm console.log, fix colors bug

* rm useless type and move children movement to reducer

* AllKeys doc

* rm null check and properly fix it

AjobK committed:
fix break colors

AjobK committed:
lemon icon changed for breaks

AjobK committed:
fix whitespace issue on expanded tt view

AjobK committed:
make props more consistent from serializer to preprocess

AjobK committed:
Only allow left click on unscheduled entries (#7449)

AjobK committed:
Hide undo/redo buttons + disable key bindings (#7448)

AjobK committed:
changed serializer key names to reflect pythonic style, using mapper in the preprocess to handle the data

renefs committed:
Add double-click functionality to expand session blocks (#7459)

* Do not zoom in if it is a poster session block

* Close expanded session block with Esc key

* Reformat button in toolbar

* Disable double click on child contribution

* Handle ESC key when modal is open

* Add title to close button in session block toolbar

* Add ESC key hint to close button in session block toolbar

* Add 'or press'

* Move kbd inline style to the toolbar scss module

AjobK committed:
refactor mappers to be even more generic and allow any new future mappers for example for sessions. change the way we handle children in serializer data to reflect more realistic api case

AjobK committed:
change old mappers to new mappers for sessions and entries, add person links mapper and use it right away

AjobK committed:
fix tsc issues, expand personlink interface

AjobK committed:
fix tests, introduce mapper for locationdata

AjobK committed:
rm console.log

ThiefMaster committed:
Fix issues when adding persons to contribs (#7437)

* Do not propagate partial to nested schemas

* Fix DB error when adding more persons to contrib

This was broken because `object` was missing in the context so the
person link schema never tried to get an existing person link.

AjobK committed:
additional personlink fields

AjobK committed:
inline date assignment

AjobK committed:
cleanup useless management/non-management stuff from serializer and add session mapper tests (personlink and location is done by default through the entry mapper)

AjobK committed:
fix biome issue

AjobK committed:
unbehead serializer

AjobK committed:
fix min duration 0 issue

AjobK committed:
fix issue where you could go 5 minutes over the bottom limit and where entries at the start would show to be 5 min later even though they scheduled on time. Took me quite a while

AjobK committed:
move new logic for limits calculation and remove margin at end of limits, which was needed before we made smarter creation logic on limits

AjobK committed:
fix start 0 bug

AjobK committed:
Add context to schema, remove any mention of conveners, fix personlinkfield to not return roles when none are passed

AjobK committed:
fix p.roles bug

ArmindoFlores committed:
Don't strip avatarURL from form in PersonDetailsModal (#7520)

ArmindoFlores committed:
Disable 'add myself' button when you added yourself (#7512)

Previously, the button was only disabled after closing and opening the
dialog again. This was because there was no userId field in persons
going through cleanupPersons().

ThiefMaster committed:
Fix isort error

ThiefMaster committed:
Fix TS error

ArmindoFlores committed:
Abort resizing when user presses ESC (#7521)

ArmindoFlores committed:
Disable drag-to-create-new when an entry is selected (#7519)

ArmindoFlores committed:
Fix contribution positioning when moving parent session block (#7513)

ThiefMaster committed:
Require a left-button click for all resize ops (#7563)

ArmindoFlores committed:
Prevent events inside entry popups from triggering in parents (#7514)

ArmindoFlores committed:
Prevent backend requests when nothing has changed (#7564)

noahsalvi committed:
Prevent clearing and incomplete values in timetable date/time pickers (#7580)

* Hide the clear button on date/time pickers

* Reject incomplete values in date/time pickers

* Make FinalDateTimePicker not be required by default

noahsalvi committed:
Display poster count on poster session blocks (#7582)

* Show poster count on poster session blocks
* Extract shared title-row spacing into SCSS variables
* Hide poster count on narrow blocks

ThiefMaster committed:
Remove undo/redo (#7591)

This would be a massive hassle during the state refactoring, and like
this it'll keep the refactoring PR cleaner.

micsucmed committed:
Hide entry move buttons when creating a draft entry (#7507)

* Hide entry move buttons when creating a draft entry

* Refactor hidden conditional

* Refacor solution not to use hidden prop

* Remove hidden prop and Omit

ArmindoFlores committed:
Wait some time before starting to drag after `mousedown` event (#7522)

renefs committed:
Add toasts to timetable via context (#7486)

Co-authored-by: AjobK <ajob.edward.kustra@cern.ch>

ThiefMaster committed:
Enable build CI job

AjobK committed:
individual timetable and contrib page fixes. partial changes sent, description attribute set with underscore and min max dates (#7614)

AjobK committed:
temporary fix for closing session modal

ArmindoFlores committed:
Show drop target overlay on drag over (#7587)

ArmindoFlores committed:
Fix UI regression when displaying unscheduled entries (#7632)

Fix regression caused by #7587 where unscheduled entries in the
side panel had their text centered instead of aligned to the left

AjobK committed:
remove css for fullscreen that makes it look like a modal

marinaej committed:
Change Side Panel UI (#7500)

* add modifications to side panel

* Fix Sidepanel regressions post rebase

* Fix lint & ruff issues

* Remove comments and unused variables

* Fix delete_timetable_entry

* Fix regressions

* Fix post review

* Fix session seletion and add action prompt component

* Fix white space

* Remove important css and fix modal submit issues

* fix drag issue

* fix actions hover issue, allow selecting text

* fix alignment and spacing in filter counter and clear cross

* remove import

* Fix tab rail size

* Your Sessions renamed to Sessions

* change create contrib button to our own button usage without wrapper

* Change ActionPrompt to ConfirmDialog and modify design

* make edit of unscheduled contributions stateful instead of location.reload based, and modify delete to only pass id

* make button text consistent, change styling and remove redundant css

* center cross in filter clear

* css fixes to sidepanel. removed absolutes, fixed overflow bugs and removed sticky which is replaced with dom hierarchy

* add gradients to sidepanel

* visual modification on session elements, including making them more compact

* minor spacing adjustments to session content

* add tooltip explaining sessions

* fix session color width bug

* expanded clarification of session

* fix regression in creation of unsch contribs

* fix filters and clear alignment regression

* rm filter tooltip

* Review fixes

* Fix regressions

* Change icon and session text

* Fix session block bug

* Fix regression with session deletion

* fix style regression on filter badge and change sessions dropdown allignment

* Fix filter element length

* change minimum width of sidebar and initial width

* tab rail spacings

* fix regression on delete session unscheduled

* fix contribs with assigned sessions not being actually unassigned when deleting session

* rename files

* #eee > rgba

* max resize handler width

* re-ordered css due to strange CSS highlighting

* Fix filter ordering alpha

---------

Co-authored-by: AjobK <ajob.edward.kustra@cern.ch>

AjobK committed:
Hide Jinja sidepanel on new timetable page (#7642)

* toggle jinja sidepanel without rerender (temporary but we could have such a thing for all management pages

* changed styling of button to be more obvious, removed footer from timetable page and removed random spacings around

micsucmed committed:
Consistently move cursor and entry edge when resizing (#7634)

Also fixes cursor blinking when resizing

AjobK committed:
Fix hiding main sidemenu (#7645)

* toggle jinja sidepanel without rerender (temporary but we could have such a thing for all management pages

* changed styling of button to be more obvious, removed footer from timetable page and removed random spacings around

* CSP support, no global vars, css moved to timetable.scss, button width fix

* rm hoisting, directly use arrow fn, adapt spacing in ::before of icon

* use event target instead of queryselect

micsucmed committed:
Add auto scroll limits on event boundaries (#7635)

ThiefMaster committed:
Make scrollable menu nicer

ThiefMaster committed:
Improve session picker in block creation

AjobK committed:
AM/PM or HH:mm depending on locale for time gutter

marinaej committed:
Add indicator for time when hovering timetable (#7646)

* Add indicator for time when hovering timetable

* Fix design after Ajob's review

* Review fixes for mouse move function

* Ts error fix

AjobK committed:
Make timetable toolbar more space-efficient (#7641)

* resized days toolbar

* rm menu.item semantic from toolbar, unpublished button smaller

* change button to be full size

* change locale for days, merge two top bars, less spacing

* reworked look of session block view toolbar

* ellipsis in session block view

* mv publication button to sidepanel

* fix scrollable days spacing issue

ThiefMaster committed:
Use locale-based 24h format in more places

ThiefMaster committed:
Fix another hardcoded en locale

ThiefMaster committed:
Revert incorect change to TT entry deletion (#7648)

There is no concept of unscheduled timetable entries on the backend, an
entry is always deleted when it's removed from the timetable. The
underlying object is left untouched during this operation, unless it's a
break which cannot exist on its own,

micsucmed committed:
Ensure proper ResizeObserver cleanup (#7669)

micsucmed committed:
Improve initial timetable scroll position (#7671)
